marker_api/src/lib.rs

```diff
@@ -32,12 +31,12 @@ pub use interface::{LintPassInfo, LintPassInfoBuilder};
 pub trait LintPass {
```

Comment:
Since there is a separation between an early and late lint pass in the rustic driver, maybe it makes sense to also name this trait `LateLintPass` and then add an `EarlyLintPass` when support for pre-macro-expanded lints is added, or how do you think this will be handled?

Reply:
Yeah, macros are something which I've been dreading for a long time. They have two big problems:
1. The question of how they should be handled/represented
2. Rustc as very limited support for them. The mentioned pre expansion pass has been softly deprecated for quite some time. I also prefer the late lint pass by a long shot.

I've extracted this topic into https://github.com/rust-marker/design/issues/47 where I described how I think we could macros. None of this is set in stone, and I would really like to have more macro-specific use cases, before making any decision on this.

If Marker grows more, it's likely that we can make supporting changes in rustc. Clippy would also benefit from better macro support. But such a change would also require several design discussions with the compiler team.

---

I assume that you asked this question in the context of breaking changes, with the question, if we should rename this trait as well. The short answer is, that I would prefer having a single lint pass for Marker, which would mean that the name can stay as it is :)